lib/mutant/expression/namespace.rb

```diff
@@ -2,24 +2,12 @@ module Mutant
   class Expression
     # Abstract base class for expressions matching namespaces
     class Namespace < self
-      include AbstractType
-
-    private
-
-      # Return matched namespace
-      #
-      # @return [String]
-      #
-      # @api private
-      #
-      def namespace
-        match[__method__]
-      end
+      include AbstractType, Anima.new(:namespace)
+      private(*anima.attribute_names)
 
       # Recursive namespace expression
       class Recursive < self
-
-        register(/\A(?<namespace>#{SCOPE_PATTERN})?\*\z/)
+        REGEXP = /\A(?<namespace>#{SCOPE_PATTERN})?\*\z/.freeze
```

Comment:
Why not name this `<scope_name>` or name the instances of `scope_name` as `namespace` ? This might allow you to share some parts of the regexp between subclasses.